src/compiler/irgen.jl

```diff
@@ -150,25 +223,45 @@ function irgen(ctx::CompilerContext)
                 LLVM.name!(llvmf, llvmfn′)
                 llvmfn = llvmfn′
             end
+
+            # finally, make function names safe for ptxas
+            # potentially unnecessary Cuda-related transform
+            llvmfn′ = safe_fn(llvmfn)
+            if llvmfn != llvmfn′
+                LLVM.name!(llvmf, llvmfn′)
+                llvmfn = llvmfn′
+            end
         end
     end
 
+    # add the global exception indicator flag
+    emit_exception_flag!(mod)
+
     # rename the entry point
-    llvmfn = replace(LLVM.name(entry), r"_\d+$"=>"")
+    if job.name !== nothing
+        llvmfn = safe_fn(string("julia_", job.name))
+    else
+        llvmfn = replace(LLVM.name(entry), r"_\d+$"=>"")
+    end
+
     ## append a global unique counter
     global globalUnique
     globalUnique += 1
     llvmfn *= "_$globalUnique"
     LLVM.name!(entry, llvmfn)
 
     # minimal required optimization
-    ModulePassManager() do pm
-        global global_ctx
-        global_ctx = ctx
+    @timeit to[] "rewrite" ModulePassManager() do pm
+        global current_job
+        current_job = job
+
+        linkage!(entry, LLVM.API.LLVMExternalLinkage)
+        internalize!(pm, [LLVM.name(entry)])
 
         add!(pm, ModulePass("LowerThrow", lower_throw!))
+        add!(pm, FunctionPass("HideUnreachable", hide_unreachable!))
```

Comment:
I don't believe either of these additions is desired, as we don't use PTXAS (which is what has problems with unreachable statements and traps); we support these just fine for debugging purposes.

Reply:
Sure. I have little knowledge of the underlying backend stuff. Just hoping to get back to regular Julia land asap :sweat_smile: